### ISSUE TITLE
DyNeRF 데이터셋 성능관련

### ISSUE THREAD

**Issue by Jho-Yonsei** [JeongminB/E-D3DGS#14]
안녕하세요?

이번 E-D3DGS 논문과 성능 관련하여 여쭙고 싶은 것이 있어 이슈란에 남깁니다.

앞서 JiaoChenGuang-JiaoHan님께서 올려주신 이슈와 같은 내용인데, ```flame_steak``` scene의 논문 내의 성능과 제가 돌려본 성능이 차이가 너무 심하게 나는 현상이 발견이 되었습니다. 

논문 내의 성능:
PSNR  SSIM
33.57  0.964

직접 얻은 성능:
PSNR  SSIM
30.55  0.948
(-3.02) (-0.016)

단순한 discrepancy라고 보기에는 PSNR은 3이나 차이가 나고, 이 결과는 비교모델들 중 가장 낮은 성능인 MixVoxel보다도 더 낮습니다. 따라서 이 부분에 관련한 질문을 몇가지 드리고싶습니다.

1. 원래 4D-Gaussian 쪽 모델들이 이렇게 성능의 fluctuation이 심한지?
2. 여러번 실험을 돌려나온 최대의 성능을 기입하신건지?

이 부분들이 궁금합니다. 또한, JiaoChenGuang-JiaoHan님께서 올려주신 이슈에는 pretrained weight를 올리는것만으로 갈무리를 지으신 것 같은데, 저 정도로 심하게 차이나는 성능이라면 이 부분에 대한 분명한 clarification이 필요해보입니다.

감사합니다.

**Comment by JeongminB** [comment]
I believe the performance degradation is due to the change in the hyperparameter value with `densify_grad_threshold` in its name. I have updated the config to use the same value as the pretrained checkpoint. The `flame_steak` scene seems to be particularly sensitive to changes in this value.

In my experience, the `flame_steak` scene's test view sometimes has large floaters near the camera. In such cases, there can be a significant performance drop even if the visual quality is good.

**Comment by seoha-kim** [comment]
And it would be much better to **translate this issue in English** considering someone who is not from South Korea. Good luck!

**Comment by LabubuJ** [comment]
> I believe the performance degradation is due to the change in the hyperparameter value with `densify_grad_threshold` in its name. I have updated the config to use the same value as the pretrained checkpoint. The `flame_steak` scene seems to be particularly sensitive to changes in this value.
> 
> In my experience, the `flame_steak` scene's test view sometimes has large floaters near the camera. In such cases, there can be a significant performance drop even if the visual quality is good.

Thanks for your great work. In other similar N3DV scenarios, this issue also persists. Despite adjusting the densify_grad_threshold, the results remain unsatisfactory. Do you have a more thorough solution to this problem?

**Comment by seoha-kim** [comment]
I've retrained with the same code and get the below results:

Method: ours_80000
Metric evaluation progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 300/300 [00:01<00:00, 228.02it/s]
Scene:  output/dynerf/flame_steak SSIM :    0.9629251
Scene:  output/dynerf/flame_steak PSNR :   33.1416969

So it's quite weird. We didn't try multiple training and write the best results.

We think PSNR tends to fluctuate with deformable 3DGS-based methods.
It's a chronic problem on 3DGS-based method, but I believe it is worse on deformation-based 3DGS than 4DGS. 

I uploaded COLMAP folder and initial points that I used to train flame steak:
https://drive.google.com/drive/folders/1viOadYzGQQs2R_h5ccKxzrXxSKnadBEE?usp=sharing

**Comment by LabubuJ** [comment]
> I've retrained with the same code and get the below results:
> 
> Method: ours_80000 Metric evaluation progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 300/300 [00:01<00:00, 228.02it/s] Scene: output/dynerf/flame_steak SSIM : 0.9629251 Scene: output/dynerf/flame_steak PSNR : 33.1416969
> 
> So it's quite weird. We didn't try multiple training and write the best results.
> 
> We think PSNR tends to fluctuate with deformable 3DGS-based methods. It's a chronic problem on 3DGS-based method, but I believe it is worse on deformation-based 3DGS than 4DGS.
> 
> I uploaded COLMAP folder and initial points that I used to train flame steak: https://drive.google.com/drive/folders/1viOadYzGQQs2R_h5ccKxzrXxSKnadBEE?usp=sharing

Thank you for your reply—the improvement in quality from your method is truly remarkable!

Interestingly, I haven't encountered such significant PSNR fluctuations in online reconstruction methods  using 3DGS as initialization. This makes me suspect that the PSNR instability might stem from the coupling between 3D Gaussians and the deformable field: even minor adjustments in the MLP can affect the color of 3D Gaussians in front of the camera, leading to noticeable PSNR variations.

In contrast, methods like 4DGS, which rely on independent 4D Gaussian primitives, can avoid this issue by directly optimizing the parameters of 4D Gaussians in front of the camera without being affected by the MLP-induced deformations.

I’d love to hear your perspective on this!

**Comment by seoha-kim** [comment]
I agree that deformable 3DGS-based methods show some PSNR fluctuations.
I’ve been involved in online 3DGS and 4DGS research projects after working on E-D3DGS.
I think online 3DGS methods have even more fluctuations than deformable 3DGS methods.
They heavily depend on the previous frame's performance and often produce big artifacts.
In my experience, 4DGS methods provide more consistent and visually better results. (but higher memory cost)

On the other hand, I don’t think PSNR is a reliable metric for evaluating visual quality.
That’s why I usually refer to SSIM instead.
